CIME/XML/env_batch.py

```diff
@@ -224,11 +224,33 @@ def get_job_overrides(self, job, case):
             if thread_count:
                 overrides["thread_count"] = thread_count
         else:
-            total_tasks = case.get_value("TOTALPES") * int(case.thread_count)
+            total_tasks = case.get_value("TOTALPES")
             thread_count = case.thread_count
         if int(total_tasks) * int(thread_count) < case.get_value("MAX_TASKS_PER_NODE"):
             overrides["max_tasks_per_node"] = int(total_tasks)
 
+        # when developed this variable was only needed on derecho, but I have tried to
+        # make it general enough that it can be used on other systems by defining MEM_PER_TASK and MAX_MEM_PER_NODE in config_machines.xml
+        # and adding {{ mem_per_node }} in config_batch.xml
+        try:
+            mem_per_task = case.get_value("MEM_PER_TASK")
+            max_mem_per_node = case.get_value("MAX_MEM_PER_NODE")
+            mem_per_node = total_tasks
```

Comment:
And this to:
`mem_per_node = case.get_value("TOTALPES")`

Reply:
Can't do that because total_tasks may be the product of an override - for example the case.st_archive script overrides total_tasks and changes it to 1.

Reply:
How about `mem_per_node = total_tasks / thread_count` ?